crates/flux-middle/src/rty/mod.rs

```diff
@@ -203,6 +203,23 @@ pub struct GeneratorObligPredicate {
     pub output: Ty,
 }
 
+#[derive(Debug, Default, Clone)]
+pub struct AssocPredicates {
+    pub predicates: Vec<AssocPredicate>,
```

Comment:
You should make this cheaply clonable if returned from a query

```suggestion
    pub predicates: List<AssocPredicate>,
```